src/dstack/_internal/server/services/fleets.py

```diff
@@ -137,6 +140,27 @@ async def get_plan(
 ) -> FleetPlan:
     # TODO: refactor offers logic into a separate module to avoid depending on runs
 
+    if spec.configuration.ssh_config and spec.configuration.ssh_config.hosts:
+        # there are manually listed hosts, need to check them for existence
+        active_instances = await list_active_remote_instances(session=session)
+
+        existing_hosts = set()
+        for instance in active_instances:
+            instance_conn_info = RemoteConnectionInfo.parse_raw(
+                cast(str, instance.remote_connection_info)
+            )
+            existing_hosts.add(instance_conn_info.host)
+
+        instances_already_in_fleet = []
+        for new_instance in spec.configuration.ssh_config.hosts:
+            if new_instance in existing_hosts:
+                instances_already_in_fleet.append(new_instance)
+
+        if instances_already_in_fleet:
+            raise ConfigurationValidationError(
+                detail=f"Instances [{', '.join(instances_already_in_fleet)}] are already assigned to a fleet."
+            )
```

Comment:
Please raise `ServerClientError("error message")`. It will be converted to a 400 response and handled correctly by the CLI.